auditbeat/module/auditd/audit_linux.go

```diff
@@ -204,9 +207,107 @@ func (ms *MetricSet) Run(reporter mb.PushReporterV2) {
 			}
 		}()
 	}
+
+	if ms.config.Reload.Enabled {
+		ruleReloader := newRuleReloader(ms.config.RuleFiles, ms.config.Reload)
+
+		go ruleReloader.Run(ms, reporter)
+	}
+
 	wg.Wait()
 }
 
+type reloader struct {
+	reload reload
+	pathes []string
+	done   chan struct{}
+	wg     sync.WaitGroup
+}
+
+func newRuleReloader(pathes []string, reload reload) *reloader {
+
+	newPathes := []string{}
+
+	for _, path := range pathes {
+		if !filepath.IsAbs(path) {
+			path = paths.Resolve(paths.Config, path)
+		}
+		newPathes = append(newPathes, path)
+	}
+
+	return &reloader{
+		reload: reload,
+		pathes: newPathes,
+		done:   make(chan struct{}),
+	}
+}
+
+// Run runs the reloader
+func (rl *reloader) Run(ms *MetricSet, reporter mb.PushReporterV2) {
+	ms.log.Info("Auditd Rule reloader started")
```

Comment:
I think it would be useful for debugging purposes to also log when it exits.

```
ms.log.Info("audit_rule_files reload watcher started")
defer ms.log.Info("audit_rule_files reload watcher stopped")
```